lib/credentials.js

```diff
@@ -70,6 +83,12 @@ Credentials.prototype._expand = function() {
     this.requestPubKey = requestDerivation.publicKey.toString();
   }
   var network = WalletUtils.getNetworkFromXPubKey(this.xPubKey);
+  if (typeof this.externalSource == "string") {
+    var xPrivKey = new Bitcore.HDPrivateKey(network);
```

Comment:
You mean we should do `var xPrivKey = new Bitcore.PrivateKey(network);` ?
About the signing from the master xpriv, should we also implement that in this pull request?